test/services/ArrivalMessageSummaryServiceSpec.scala

```diff
@@ -0,0 +1,219 @@
+ */
+
+package services
+
+import java.time.LocalDateTime
+
+import base.SpecBase
+import cats.data._
+import generators.ModelGenerators
+import models.MessageStatus._
+import models.Arrival
+import models.MessageId
+import models.MessageStatus
+import models.MessageType
+import models.MovementMessage
+import models.MovementMessageWithStatus
+import models.MovementMessageWithoutStatus
+import org.scalacheck.Arbitrary.arbitrary
+import models.MessageType._
+import org.scalacheck.Gen
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+
+class ArrivalMessageSummaryServiceSpec extends SpecBase with ModelGenerators with ScalaCheckDrivenPropertyChecks {
+  import ArrivalMessageSummaryServiceSpec.MovementMessagesHelpers._
+
+  def messageGeneratorSent(messageType: MessageType): Gen[MovementMessageWithStatus] = {
+    val message = xml.XML.loadString(s"<${messageType.rootNode}>test</${messageType.rootNode}>")
+    arbitrary[MovementMessageWithStatus].map(_.copy(messageType = messageType, message = message, status = SubmissionPending))
+  }
+
+  def messageGeneratorResponse(messageType: MessageType): Gen[MovementMessageWithoutStatus] = {
+    val message = xml.XML.loadString(s"<${messageType.rootNode}>test</${messageType.rootNode}>")
+    arbitrary[MovementMessageWithoutStatus].map(_.copy(messageType = messageType, message = message))
+  }
+
+  val ie007Gen = messageGeneratorSent(ArrivalNotification)
+  val ie008Gen = messageGeneratorResponse(ArrivalRejection)
+
+  def arrivalMovement(msgs: NonEmptyList[MovementMessage]): Gen[Arrival] =
+    for {
+      arrival <- arbitrary[Arrival]
+    } yield arrival.copy(messages = msgs)
+
+  "IE007" - {
+
+    "must return the original IE007 when there have been no other messages" in {
+      val service = new ArrivalMessageSummaryService
+
+      forAll(ie007Gen) {
+        ie007 =>
+          forAll(arrivalMovement(NonEmptyList.one(ie007))) {
+            arrival =>
+              val (message, messageId) = service.arrivalNotification(arrival)
+
+              message mustEqual ie007
+              messageId mustEqual MessageId.fromMessageIdValue(1).value
+
+          }
+      }
+    }
+
+    "must return the original IEOO7 and first IE008 when there is only an IE007 and a IE008" in {
+      val service = new ArrivalMessageSummaryService
+
+      forAll(ie007Gen, ie008Gen) {
+        (ie007, ie008) =>
+          val messages = NonEmptyList.of(ie007, ie008)
+
+          forAll(arrivalMovement(messages)) {
+            arrival =>
+              val (message, messageId) = service.arrivalNotification(arrival)
+
+              message mustEqual ie007
+              messageId mustEqual MessageId.fromMessageIdValue(1).value
+          }
+      }
+
+    }
+
+    "must return the new IEOO7 when there has been a correction to a rejected arrival message" in {
+      val service = new ArrivalMessageSummaryService
+
+      forAll(ie007Gen.submitted.msgCorrId(1), ie008Gen.msgCorrId(1), ie007Gen.msgCorrId(2)) {
```

Comment:
Does this mean we can set values manually here rather than taking the generated value and overriding values with _.copy()_

Reply:
Under the hood it does still call copy, it just makes it easier to write, https://github.com/hmrc/transit-movements-trader-at-destination/pull/115/files/f8728c3b88ace423fe860c14d275e9330acc07f4#diff-ae0ea5187d42d5f23a9b45db82935870R209-R211

Reply:
Nice one 👍